ql/time/asx.cpp

```diff
@@ -57,109 +60,60 @@ namespace QuantLib {
         if (in.length() != 2)
             return false;
 
-        string str1("0123456789");
-        string::size_type loc = str1.find(in.substr(1,1), 0);
-        if (loc == string::npos)
+        // 2nd character of code needs to be digit
+        if (!std::isdigit(in[1]))
```

Comment:
The argument to `std::isdigit` should always be cast to `unsigned char` first. See https://en.cppreference.com/w/cpp/string/byte/isdigit.